variation-commons-core/src/test/java/uk/ac/ebi/eva/commons/core/models/factories/VariantVcfFactoryTest.java

```diff
@@ -649,42 +637,14 @@ public void multiallelicVariantAndOneAlleleHasNoGenotypes() {
     }
 
     @Test
-    public void variantWithAlleleFrequencyZero() {
-        String line = "1\t10040\trs123\tT\tC\t.\t.\tAF=0";
-
-        thrown.expect(NonVariantException.class);
-        factory.create(FILE_ID, STUDY_ID, line);
-    }
-
-    @Test
-    public void multiallelicWithAlleleFrequencyZero() {
-        String line = "1\t10040\trs123\tT\tC,G,A\t.\t.\tAF=0.5,0,0.2";
-
-        thrown.expect(NonVariantException.class);
-        factory.create(FILE_ID, STUDY_ID, line);
-    }
-
-    @Test
-    public void variantWithAlleleCountZero() {
-        String line = "1\t10040\trs123\tT\tC\t.\t.\tAC=0";
-
-        thrown.expect(NonVariantException.class);
-        factory.create(FILE_ID, STUDY_ID, line);
-    }
-
-    @Test
-    public void multiAlleleWithOneAlleleCountZero() {
-        String line =  "1\t10040\trs123\tT\tC,G,A\t.\t.\tAC=0.5,0,0.2";
-
-        thrown.expect(NonVariantException.class);
-        factory.create(FILE_ID, STUDY_ID, line);
-    }
-
-    @Test
-    public void variantWithAlleleTotalNumberZero() {
-        String line = "1\t10040\trs123\tT\tC\t.\t.\tAN=0";
+    public void infoAttributesWontBeValidated() {
+        List<Variant> expResult = new LinkedList<>();
+        expResult.add(new Variant("1", 1000, 1000, "T", "G"));
+        String line;
 
-        thrown.expect(NonVariantException.class);
-        factory.create(FILE_ID, STUDY_ID, line);
+        // the variant is valid if it has some alternate genotypes, even if AF, AC or AN are zero
+        line = "chr1\t1000\t.\tT\tG\t.\t.\tAF=0;AC=0;AN=0\tGT\t0/1";
+        List<Variant> result = factory.create(FILE_ID, STUDY_ID, line);
+        assertEquals(expResult, result);
     }
 }
```

Comment:
I have gone through all the tests in this class and couldn't find any that check an exception if thrown if all the genotypes are reference or missing. There is one with a bunch of missing genotypes but one is not missing.

Reply:
the method "variantWithNoAltGenotypes" does that, and there are two more similar methods for haploid and triploid variants. I will add two more methods, one for testing a variant just with missing values, and one just with 0/0 calls.